lib/Doctrine/DBAL/Connection.php

```diff
@@ -771,14 +769,12 @@ public function fetchAll($sql, array $params = array(), $types = array())
      *
      * @param string $statement The SQL statement to prepare.
      *
-     * @return \Doctrine\DBAL\Driver\Statement The prepared statement.
+     * @return \Doctrine\DBAL\Statement The prepared statement.
      *
      * @throws \Doctrine\DBAL\DBALException
      */
     public function prepare($statement)
     {
-        $this->connect();
```

Comment:
are you sure about that ? I'm quite sure that preparing a statement requires being connected